contracts/FastBreakV1.cdc

```diff
@@ -672,41 +686,66 @@ pub contract FastBreakV1: NonFungibleToken {
             destroy tokens
         }
 
-        pub fun getIDs(): [UInt64] {
+        view access(all) fun getIDs(): [UInt64] {
             return self.ownedNFTs.keys
         }
 
-        pub fun borrowNFT(id: UInt64): &NonFungibleToken.NFT {
-            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)!
+        view access(all) fun borrowNFT(_ id: UInt64): &{NonFungibleToken.NFT}? {
+            return &self.ownedNFTs[id]
         }
 
-        pub fun borrowNFTSafe(id: UInt64): &NonFungibleToken.NFT? {
-            return (&self.ownedNFTs[id] as &NonFungibleToken.NFT?)
+        view access(all) fun borrowNFTSafe(id: UInt64): &{NonFungibleToken.NFT}? {
+            return &self.ownedNFTs[id]
         }
 
-        pub fun borrowFastBreakNFT(id: UInt64): &FastBreakV1.NFT? {
-            if self.ownedNFTs[id] != nil {
-                let ref = (&self.ownedNFTs[id] as auth &NonFungibleToken.NFT?)!
-                return ref as! &FastBreakV1.NFT
-            } else {
-                return nil
+        view access(all) fun borrowFastBreakNFT(id: UInt64): &FastBreakV1.NFT? {
+            return self.borrowNFT(id) as! &FastBreakV1.NFT?
+        }
+
+        view access(all) fun getSupportedNFTTypes(): {Type: Bool} {
+            let supportedTypes: {Type: Bool} = {}
+            supportedTypes[Type<@FastBreakV1.NFT>()] = true
+            return supportedTypes
+        }
+
+        // Return whether or not the given type is accepted by the collection
+        // A collection that can accept any type should just return true by default
+        view access(all) fun isSupportedNFTType(type: Type): Bool {
+            if type == Type<@FastBreakV1.NFT>() {
+                return true
             }
+            return false
         }
 
-        destroy() {
-            destroy self.ownedNFTs
+        access(all) fun createEmptyCollection(): @{NonFungibleToken.Collection} {
+            return <- FastBreakV1.createEmptyCollection(nftType: Type<@FastBreakV1.NFT>())
+        }
+
+        view access(all) fun getLength(): Int {
+            return self.ownedNFTs.keys.length
         }
 
         init() {
             self.ownedNFTs <- {}
         }
     }
 
-    pub fun createEmptyCollection(): @NonFungibleToken.Collection {
+    access(all) fun createEmptyCollection(nftType: Type): @{NonFungibleToken.Collection} {
+        if nftType != Type<@FastBreakV1.NFT>() {
+            panic("NFT type is not supported")
+        }
         return <- create Collection()
     }
 
-    pub fun createPlayer(playerName: String): @FastBreakV1.Player {
+    view access(all) fun getContractViews(resourceType: Type?): [Type] {
+        return []
+    }
+
+    view access(all) fun resolveContractView(resourceType: Type?, viewType: Type): AnyStruct? {
+        return nil
+    }
```

Comment:
I would think that there would be some views that this contract would implement. You should figure out which ones can be included so Top Shot can set a good example